### DISCUSSION TITLE
Copy types of `AppRouter` to different file

### DISCUSSION THREAD

**Discussion by IgnisDa** [trpc/trpc#1683] [Q&A] [answered]
First of all thank you for this awesome tool!

I am using [NX](https://nx.dev) to manage my monorepo. With the current setup of trpc client, I will have to directly import the `AppRouter` type from the backend into the frontend. This will introduce a direct dependency between the backend and frontend which I don't want. 

Instead, I want to write a script that can copy the types of `AppRouter` to a new file that I can import in the frontend. I will periodically run this script to keep the frontend and backend in sync.

Any pointers on how I should go about doing this?

**Comment by ecwyne** [comment, marked as answer]
As of Typescript 3.8 you can use `import type` to only import types. https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export

**Reply by tkroll** [reply]
Good afternoon, answered question.

Been trying to import the AppRouter type from my server app to a few client apps in NX. 
What's the suggested way to get around `ESLint: Imports of apps are forbidden(@nrwl/nx/enforce-module-boundaries)`

Loving tRPC! Thanks

**Reply by jlalmes** [reply]
Hi @tkroll, here are a few options: 
  1. Create a shared library (i.e. `libs/routers`) that you can `import` from your server and `import type` from your clients. 
  2. Modify module boundary rule constraints, read more: https://nx.dev/structure/monorepo-tags
  3. Just add this above your import `/* eslint-disable-next-line @nrwl/nx/enforce-module-boundaries */`

**Reply by tkroll** [reply]
Thanks will give all three a try.

**Reply by Pruxis** [reply]
for me importing wasn't a problem at all, but what happends is the client doesn't take into account type changes on the fly.
I've tried fiddling with the project implicitDependencies + tsconfig files, now looking into webpack if it's not being included there smh
Maybe any of you already had this issue and solved it

**Reply by tkroll** [reply]
> for me importing wasn't a problem at all, but what happends is the client doesn't take into account type changes on the fly. I've tried fiddling with the project implicitDependencies + tsconfig files, now looking into webpack if it's not being included there smh Maybe any of you already had this issue and solved it

Are you using NX? Creating a shared library worked for me and the client gets the changes.
